### ISSUE TITLE
doxygen: Corrupted html in template param list when using pointer value param with default value

### ISSUE THREAD

**Issue by marzer** [mosra/m.css#74]
*m.css*: current master
*doxygen*: 1.8.14
*python*: 3.7

Sorry for the wordy title. Essentially, given this code:

```cpp
	/// \brief	More readable alias of std::enable_if_t.
	template <bool V, typename U = void>
	using OnlyIf = std::enable_if_t<V, U>;

	/// \brief	Unsubscribes all class member functions sharing the same callee.
	///
	/// \tparam	CALLEE	The callee class type.
	/// \param	callee	The callee.
	///
	/// \return	A reference to the Event.
	template <typename CALLEE, OnlyIf<
		!std::is_function_v<CALLEE>
		&& std::is_class_v<CALLEE>
	>* = nullptr>
	auto& operator-= (CALLEE* callee) noexcept
	{
		/* .... */
		return *this;
	}
```

*m.css* produces this output:
![m css-output](https://user-images.githubusercontent.com/3958019/48728863-d9bd7600-ec3e-11e8-9de9-98a9ce17e4e5.PNG)

When vanilla doxygen seems to handle it OK:
![doxygen-output](https://user-images.githubusercontent.com/3958019/48728889-f063cd00-ec3e-11e8-837a-21bf1e9f0d72.PNG)

Here's the xml used to generate the m.css version (renamed as .txt because github is fussy):
[_event_8h.txt](https://github.com/mosra/m.css/files/2596696/_event_8h.txt)

The issue doesn't occur if I instead do this:
```cpp
template <typename T, typename = OnlyIf</*...*/>>
```
But in this particular case I already have a competing overload using that form, and can't start adding dummy parameters etc, since it's an operator. The issue also isn't isolated to this specific piece of code, either; It happens whenever `* = nullptr>` appears at the end of a template argument list.

Let me know if you need more info ^_^ 🎉

**Comment by mosra** [comment]
Hi!

Wtf. Not sure what happens here. I hope it doesn't require me parsing the template arguments to fix this, because if there's one thing I don't want to do, it's parsing C++ code :laugh:

The attached file doesn't contain the operator documentation, just one file-specific macro. The function output should be in some `class_achilles_event.xml` file (aka the XML file for the class, not for the header file it's defined in). Can you attach that file instead? Thank you! :+1:

**Comment by marzer** [comment]
Oh, duh. That was silly of me. Will get you that file in a bit.

**Comment by marzer** [comment]
Here you go: 
[class_achilles_1_1_event.txt](https://github.com/mosra/m.css/files/2610712/class_achilles_1_1_event.txt)

Line 527 is the troublemaker, methinks.

**Comment by mosra** [comment]
Hi, this particular case should be fixed in 61b97022ae6ac0944bfbc9283a2d500693587ee4. I used your (reduced) input as a test case, nevertheless please complain if it's not working as it should be.

Doing more involved stuff *of course* requires parsing C++, so if you'd have function pointers as template arguments, it will most probably not behave correctly.

**Comment by marzer** [comment]
Ah, awesome! Works perfectly, cheers ^_^
